gateway/ln-gateway/src/lib.rs

```diff
@@ -1195,26 +1175,48 @@ impl Gateway {
                             .await
                             {
                                 Err(GatewayError::FederationError(FederationError::general(
-                                    anyhow::anyhow!("Error registering federation {federation_id}: {e:?}")
+                                    anyhow::anyhow!(
+                                        "Error registering federation {federation_id}: {e:?}"
+                                    ),
                                 )))?
                             }
-                        } else {
-                            Err(GatewayError::FederationError(FederationError::general(
-                                anyhow::anyhow!("Could not retrieve federation config for {federation_id}")
-                            )))?
                         }
                     }
                 }
-                Err(e) =>
-                    Err(GatewayError::LightningRpcError(LightningRpcError::FailedToGetRouteHints {
-                        failure_reason: format!("Could not retrieve route hints, gateway will not be registered for now: {e:?}")
-                    }
-                    ))?
+                Err(e) => Err(GatewayError::LightningRpcError(
+                    LightningRpcError::FailedToGetRouteHints {
+                        failure_reason: format!(
+                            "Could not retrieve route hints, gateway will not be registered for now: {e:?}"
+                        ),
+                    },
+                ))?,
             }
         }
         Ok(())
     }
 
+    /// Iterates through all of the federation configurations and registers the
+    /// gateway with each federation.
+    async fn register_all_federations(
+        gateway: &Gateway,
+        gateway_config: &GatewayConfiguration,
+    ) -> Result<()> {
+        let mut all_federations: Vec<(FederationId, FederationConfig)> = Vec::new();
+        let mut dbtx = gateway.gateway_db.begin_transaction_nc().await;
```

Comment:
A more compact way of doing the same thing is something like this:

```
dbtx.find_by_prefix(&FederationIdPrefix).await.collect::<(FederationId, FederationConfig)>().await
```

You don't need to lookup each config individually. Then you can delete this function since its only 1 line to collect all of the federations, and just call `register_federations` instead